### PR TITLE
Query set type feature

### PR DESCRIPTION
**Pull request by ger-benjamin**
Fix: #2604 (query groups)
Fix: #2672 (ngeo query examples broken)
Fix: #2844 (fix examples broken by permalink)
Also for GEO-471
And update ol3 (to a tag)

Example: https://testgmf.sig.cloud.camptocamp.net/bge

also (for grid): https://camptocamp.github.io/ngeo/query_set_type_feature/examples/contribs/gmf/apps/desktop_alt/index.html?debug&lang=en&baselayer_ref=OSM%20map%20WMS&tree_groups=Layers%2CGroup&tree_group_layers_Transport=fuel&map_x=615823&map_y=177798&map_zoom=4&tree_group_opacity_Group=0.65&tree_group_layers_Group=osm&tree_group_layers_Layers

or other changes like debug examples: https://camptocamp.github.io/ngeo/query_set_type_feature/examples/bboxquery.html
...

### CODE REVIEW

File: src/services/mapquerent.js [hunk 1]
```diff
@@ -174,17 +174,29 @@ ngeo.MapQuerent = class {
...
+        if (!featuresByType[feature.type]) {
+          featuresByType[feature.type] = [];
+        }
+        featuresByType[feature.type].push(feature);
       });
-      total += features.length;
+
+      for (const type in featuresByType) {
+        label = type ? type : label;
+        const filteredFeatures =  featuresByType[type];
```
Comment (ger-benjamin) [inline src/services/mapquerent.js]: Rename me to `typeSeparatedFeatures` or eq.

**Comment by fredj** [comment]
@ger-benjamin OpenLayers PR merged

**Comment by ger-benjamin** [comment]
@fredj Done, examples in description. Can you review that ?

File: src/services/featurehelper.js [hunk 2]
```diff
@@ -603,14 +603,16 @@ ngeo.FeatureHelper.prototype.getHaloStyle_ = function(feature) {
 /**
  * Delete the unwanted ol3 properties from the current feature then return the
  * properties.
+ * Delete also the 'ngeo_feature_type_' from the ngeo query system.
  * @param {!ol.Feature} feature Feature.
  * @return {!Object.<string, *>} Filtered properties of the current feature.
  * @export
  */
-ngeo.FeatureHelper.prototype.getFilteredFeatureValues = function(feature) {
+ngeo.FeatureHelper.getFilteredFeatureValues = function(feature) {
```
Comment (fredj) [inline src/services/featurehelper.js]: nice simplification

**Comment by ger-benjamin** [comment]
Mhhh @fredj I can't reproduce the travis errors on my own pc...

**Comment by ger-benjamin** [comment]
@fred Travis is happy, me too. Can I merge ?

**Pull request merged**